File: cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int main(void)
{
   float change;
   do {
       change = get_float("how much change?\n");
   }
   while (change < 0);
   int changetotal = round(change * 100);
    
    
    //got change total. let's count the coins
    
    int dnponly = changetotal;
    int quarters = 0;
    
    while (dnponly > 24)
   {dnponly = dnponly - 25;
   quarters = quarters + 1;}
   
   //took out and counted quarters; dimes, nickels and pennies remain
   
   int nponly = dnponly;
   int dimes = 0;
   
   while (nponly > 9)
   {nponly = nponly - 10;
   dimes = dimes + 1; }
   
   //took out and counted dimes; nickels and pennies remain
   
   int ponly = nponly;
   int nickels = 0;
   
   while (ponly > 4)
   {ponly = ponly - 5;
   nickels = nickels +1; }
   
   //took out and counted nickels. pennies remain
   
   int only = ponly;
   int pennies = 0;
   
   while (only > 0)
   { only = only - 1;
   pennies = pennies +1; }
   
   //took out and counted all coins. Time to add up the total number of coins
   
   int total = pennies + nickels + dimes + quarters;

printf("quarters: %i\n", quarters);
printf("dimes: %i\n", dimes);
printf("nickels: %i\n", nickels);
printf("pennies: %i\n", pennies);
printf("total coins: %i\n", total);
printf("%i\n", total);
}
    
    

File: cash.py
from cs50 import get_float

# get change total from user
change = get_float("Change owed: ")
while change < 0: 
    change = get_float("Change owed: ")

# determine number of quarters
quarters = 0
while (change - .25) >= 0.00:
    quarters += 1
    change -= .25
    
# determine dimes
dimes = 0
while (change -.10) >= 0.00:
    dimes += 1
    change -= .10

# determine nickels
nickels = 0
while (round(change, 2) -.05) >= 0.00:
    nickels += 1
    change -= .05

# determine pennies
pennies = 0
while (round(change, 2) - .01) >= 0.00:
    pennies += 1
    change -= .01
    
# print total coins
total = quarters + dimes + nickels + pennies
print(total)